s3_deploy.sh

```diff
@@ -0,0 +1,85 @@
+
+#!/bin/bash
+# Project Name used as folder in S3 bucket
+PROJECT_NAME='seismic-explorer'
+# AWS CloudFront distribution ID
+DISTRIBUTION_ID='E3MT1G8AEEAXO'
+# AWS CloudFront distribution domain
+DISTRIBUTION_DOMAIN='seismic-explorer.concord.org'
+# name of branch to deploy to root of site
+ROOT_BRANCH='production'
+# location of built files
+SRC_DIR='dist'
+
+# extract current TAG if present
+# the 2> is to prevent error messages when no match is found
+CURRENT_TAG=`git describe --tags --exact-match $GITHUB_SHA 2> /dev/null`
+
+# Extract the branch or tag name from the GITHUB_REF
+# it should either be: refs/head/branch-name or
+# or refs/tags/v1.2.3
+# since we ought to know if this is a branch or tag based on the ref
+# we could simplify the CURRENT_TAG approach above
+BRANCH_OR_TAG=${GITHUB_REF#refs/*/}
+echo branch or tag: $BRANCH_OR_TAG
+
+# strip PT ID from branch name for branch builds
+DEPLOY_DIR_NAME=$BRANCH_OR_TAG
+PT_PREFIX_REGEX="^([0-9]{8,}-)(.+)$"
+PT_SUFFIX_REGEX="^(.+)(-[0-9]{8,})$"
+if [[ $DEPLOY_DIR_NAME =~ $PT_PREFIX_REGEX ]]; then
+  DEPLOY_DIR_NAME=${BASH_REMATCH[2]}
+fi
+if [[ $DEPLOY_DIR_NAME =~ $PT_SUFFIX_REGEX ]]; then
+  DEPLOY_DIR_NAME=${BASH_REMATCH[1]}
+fi
+
+# tagged builds deploy to /version/TAG_NAME
+if [ "$BRANCH_OR_TAG" = "$CURRENT_TAG" ]; then
+  mkdir -p _site/version
+  S3_DEPLOY_DIR="version/$BRANCH_OR_TAG"
+  DEPLOY_DEST="_site/$S3_DEPLOY_DIR"
+  INVAL_PATH="/version/$BRANCH_OR_TAG/index.html"
+  # in this case we are going to deploy this code to a subfolder of version
+  # So ignore everything except this folder.
+  # FIXME: the code used to be calling Regexp.escape on S3_DEPLOY_DIR
+  IGNORE_ON_SERVER="^(?!$PROJECT_NAME/$S3_DEPLOY_DIR/)"
+
+# root branch builds deploy to root of site
+elif [ "$BRANCH_OR_TAG" = "$ROOT_BRANCH" ]; then
+  DEPLOY_DEST="_site"
+  INVAL_PATH="/index.html"
+  # in this case we are going to deploy this branch to the top level
+  # so we need to ignore the version and branch folders
+  IGNORE_ON_SERVER="^$PROJECT_NAME/(version/|branch/)"
+
+# branch builds deploy to /branch/BRANCH_NAME
+else
+  mkdir -p _site/branch
+  S3_DEPLOY_DIR="branch/$DEPLOY_DIR_NAME"
+  DEPLOY_DEST="_site/$S3_DEPLOY_DIR"
+  INVAL_PATH="/branch/$DEPLOY_DIR_NAME/index.html"
+  # in this case we are going to deploy this code to a subfolder of branch
+  # So ignore everything except this folder.
+  # FIXME: the code used to be calling Regexp.escape on S3_DEPLOY_DIR
+  IGNORE_ON_SERVER="^(?!$PROJECT_NAME/$S3_DEPLOY_DIR/)"
+fi
+
+# used by s3_website.yml
+export PROJECT_NAME
+export IGNORE_ON_SERVER
+export DISTRIBUTION_ID
+export DISTRIBUTION_DOMAIN
+
+# copy files to destination
+mv $SRC_DIR $DEPLOY_DEST
+
+# deploy the site contents
+echo Deploying "$BRANCH_OR_TAG" to "$PROJECT_NAME/$S3_DEPLOY_DIR"...
+# temporarily add dry-run switch to make sure recent changes are doing the right thing
+s3_website push --dry-run --site _site
```

Comment:
```suggestion
s3_website push --site _site
```
I didn't remove this yet because I was worried about it messing up old versions since I made so many changes. Perhaps you want to wait until I disable this on the starter-projects before doing it here.

Reply:
@scytacki I hadn't spotted this, good to know. Yes I'd like to make sure we don't accidentally blow away all the old versions. If we're doing that we do in intentionally, not accidentally :)